src/buildCommon.js

```diff
@@ -261,45 +322,44 @@ type VListChild = VListElem | VListKern;
 type VListParam = {|
     // Each child contains how much it should be shifted downward.
     positionType: "individualShift",
-    children: (VListElem & {shift: number})[],
+    children: VListElemAndShift[],
 |} | {|
     // "top": The positionData specifies the topmost point of the vlist (note this
     //        is expected to be a height, so positive values move up).
     // "bottom": The positionData specifies the bottommost point of the vlist (note
     //           this is expected to be a depth, so positive values move down).
     // "shift": The vlist will be positioned such that its baseline is positionData
-    //          away from the baseline of the first child. Positive values move
-    //          downwards.
+    //          away from the baseline of the first child which MUST be an
+    //          "elem". Positive values move downwards.
     positionType: "top" | "bottom" | "shift",
     positionData: number,
     children: VListChild[],
 |} | {|
     // The vlist is positioned so that its baseline is aligned with the baseline
-    // of the first child. This is equivalent to "shift" with positionData=0.
+    // of the first child which MUST be an "elem". This is equivalent to "shift"
+    // with positionData=0.
     positionType: "firstBaseline",
     children: VListChild[],
 |};
-*/
 
-/**
- * Makes a vertical list by stacking elements and kerns on top of each other.
- * Allows for many different ways of specifying the positioning method.
- *
- * See parameter documentation on the type documentation above.
- */
-const makeVList = function({positionType, positionData, children}, options) {
-    let depth;
-    let currPos;
-    let i;
-    if (positionType === "individualShift") {
-        const oldChildren = children;
-        children = [oldChildren[0]];
 
-        // Add in kerns to the list of children to get each element to be
+// Computes the updated `children` list and the overall depth.
+//
+// This helper function for makeVList makes it easier to enforce type safety by
+// allowing early exits (returns) in the logic.
+const getVListChildrenAndDepth = function(params: VListParam): {
+    children: (VListChild | VListElemAndShift)[] | VListChild[],
```

Comment:
This should probably be defined as a type somewhere.  Also, what's the benefit of typing both the situation where there's a mix of `VListChild` and `VListElemAndShift` in the `children` array as well as the the case where `children` is all `VListChild`s.